src/resource_clients/request_queue.ts

```diff
@@ -234,6 +335,24 @@ export interface RequestQueueClientAddRequestResult {
     wasAlreadyHandled: boolean;
 }
 
+interface ProcessedRequest {
+    uniqueKey: string;
+    requestId: string;
+    wasAlreadyPresent: boolean;
+    wasAlreadyHandled: boolean;
+}
+
+interface UnprocessedRequest {
+    uniqueKey: string;
+    url: string;
+    method?: AllowedHttpMethods;
+}
+
+export interface RequestQueueClientBatchAddRequestsResult {
+    processedRequests: Array<ProcessedRequest>,
+    unprocessedRequests: Array<UnprocessedRequest>,
```

Comment:
we use `[]` syntax for arrays, will add that to the linter

```suggestion
    processedRequests: ProcessedRequest[],
    unprocessedRequests: UnprocessedRequest[],
```